### PR TITLE
Implement omnisharp-navigate-to-usage and omnisharp-navigate-to-implementation.

### PR DESCRIPTION
**Pull request by retran**
Usage of omnisharp-navigate-to-symbol is much more comfortable then usage of omnisharp-find-usages and omnisharp-find-implementations, so i implemented my own variants based on implementation of navigate-to-* functions.

With fuzzy matching for ido, it is now much easier to navigate through code.

screenshot from 2015-03-20 22 57 04

### CODE REVIEW

File: omnisharp.el [hunk 1]
```diff
@@ -1078,6 +1078,47 @@ cursor at that location"
         imenu-list)
     (error nil)))
 
+;; http://xahlee.blogspot.ru/2011/09/emacs-lisp-function-to-trim-string.html
+(defun trim-string (string)
```
Comment (sp3ctum) [inline omnisharp.el]: Let's remove this and use this instead:
https://github.com/magnars/s.el#s-trim-s

**Comment by sp3ctum** [comment]
Hi, thanks for the patches.
I'm not sure I understand what they are for, though. Please consider:
- there is no `omnisharp-navigate-to-symbol` like you suggest in the op
- both functions already exist, see `omnisharp-find-usages` and `omnisharp-find-implementations`

What am I missing?

**Comment by retran** [comment]
Hi!

- Sorry for mistake. Of course, I mean omnisharp-navigate-to-solution-member (with ido-mode).
- omnisharp-find-usages and omnisharp-find-implementations are very good for code exploration, but not fit very well for navigation. With Resharper`s Find Usages window I can quickly jump to the specific line using fuzzy matching (https://www.jetbrains.com/resharper/features/navigation_search.html, section "Find Usages"). We can achieve the same behavior with ido-mode (i know about helm, but i don`t like it and I think there should be a choice). Unfortunately, i don`t know how to use asynchronous requests with ido-mode, so i implemented synchronous version of "Find Usages" and "Find Implementations" that work exactly as omnisharp-navigate-to-solution-member.

I agree about trim-string (thanks for good library ;))) and will commit changes soon (with some enchancements for "find implementations" formatting).

**Comment by sp3ctum** [comment]
Ah I see. That is a great idea. If you can, submit your code and we'll see
about the asynchronous part later.
26.3.2015 13.29 kirjoitti "Andrew Vasilyev" <notifications@github.com>:

**Comment by retran** [comment]
Done :)

**Comment by sp3ctum** [comment]
Sorry, I thought I had merged this in already. I'll make time to test this and merge it in then.

**Comment by sp3ctum** [comment]
It worked nicely, thanks!

I merged it in but messed it up somehow so it doesn't show up nicely here. The changes are in, though.

**Pull request closed without merging**